worker/s3caller/client.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package s3caller
+
+import (
+	"context"
+	"io"
+	"net/http"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/juju/errors"
+	"gopkg.in/httprequest.v1"
+
+	"github.com/juju/juju/agent"
+	"github.com/juju/juju/api"
+)
+
+type Session interface {
+	GetObject(ctx context.Context, bucketName, objectName string) (io.ReadCloser, error)
+}
+
+type objectsClient struct {
+	logger Logger
+	client *s3.Client
+}
+
+func (c *objectsClient) GetObject(ctx context.Context, bucketName, objectName string) (io.ReadCloser, error) {
+	c.logger.Tracef("retrieving bucket %s object %s from s3 storage", bucketName, objectName)
+	obj, err := c.client.GetObject(ctx,
+		&s3.GetObjectInput{
+			Bucket: aws.String(bucketName),
+			Key:    aws.String(objectName),
+		})
+	if err != nil {
+		return nil, errors.Annotatef(err, "unable to get object %s on bucket %s using S3 client", objectName, bucketName)
+	}
+	return obj.Body, nil
+}
+
+type awsEndpointResolver struct {
+	endpoint string
+}
+
+func (a *awsEndpointResolver) ResolveEndpoint(service, region string) (aws.Endpoint, error) {
+	return aws.Endpoint{
+		URL: "https://" + a.endpoint,
+	}, nil
+
+}
+
+type awsHttpClient struct {
```

Comment:
This should be called `awsHTTPClient`. There is a go document on this, but I can't find it.

Reply:
changed it to `awsHTTPDoer`